### PR TITLE
Add some debugging aids

### PR DESCRIPTION
**Pull request by abarth**
I've been using these locally, but they're likely to be useful to other people
as well.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/cupertino/slider.dart [hunk 1]
```diff
@@ -111,6 +111,14 @@ class CupertinoSlider extends StatefulWidget {
 
   @override
   _CupertinoSliderState createState() => new _CupertinoSliderState();
+
+  @override
+  void debugFillDescription(List<String> description) {
+    super.debugFillDescription(description);
+    description.add('value: $value');
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/slider.dart]: please put `toStringAsFixed(1)` on any doubles in debugFillDescription.

File: packages/flutter/lib/src/rendering/object.dart [hunk 2]
```diff
@@ -2018,6 +2018,11 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
       parent.markNeedsPaint();
       assert(parent == this.parent);
     } else {
+      assert(() {
+        if (debugPrintMarkNeedsPaintStacks)
+          debugPrintStack(label: 'markNeedsPaint() called for $this');
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/object.dart]: do we want to add a parenthetical like `(root of render tree)`?

**Comment by Hixie** [comment]
LGTM

**Pull request merged**